### ISSUE TITLE
ui: some UI elements suddenly changed into weird symbols

### ISSUE THREAD

**Issue by LasTshaMAN** [decred/dcrdex#1652]
Playing around in UI (before registering with a server, trying to add testnet server) suddenly I see this (many, but not all, elements start displaying weird symbol combinations instead of English-names):

<img width="432" alt="image" src="https://user-images.githubusercontent.com/7031569/172049494-2c77141e-f628-46c8-afc4-2eb6bd05fc61.png">

<img width="432" alt="image" src="https://user-images.githubusercontent.com/7031569/172049707-aa2e7663-b538-4670-88c1-594ba7b689e2.png">

<img width="432" alt="image" src="https://user-images.githubusercontent.com/7031569/172049645-4e2fc4e8-b3a5-4b5a-94ab-1af6bf5b2d98.png">

<img width="432" alt="image" src="https://user-images.githubusercontent.com/7031569/172049733-55ff9612-f611-423d-bb65-1439423146cf.png">

screenshots above are taken in Brave browser (navigating UI without refreshing the page),  opening the same page in a different tab, opening in Firefox (and logging in) or refreshing the page turns everything back to normal,

haven't been able to reproduce 2nd time,
and didn't find anything related in `dexc` logs.

**Comment by LasTshaMAN** [comment]
At first I thought maybe English/Chinese translations somehow got mixed (I'm not familiar with Chinese, cut me some slack :), but then I've noticed that just icons changed and not text.

**Comment by ukane-philemon** [comment]
Hmm, I've noticed this severally. I think is/could be caused by poor internet connection at the time.

**Comment by chappjc** [comment]
This can only happen if the browser failed to download one of the icomoon font files:

https://github.com/decred/dcrdex/blob/a1d4690e5837c5882a26ff31034a657987bb00ba/client/webserver/site/src/css/typography.scss#L40-L45

or there's an extension blocking their use.

I've never seen it happen.  We'll need a reproduction otherwise I think this is just the browser messing up.

@martonp Note that there are cache busters on the icomoon font paths.  I always forget about those, but I think you have an active PR that adds a glyph (or did we merge it already?), so should bump that.

**Comment by LasTshaMAN** [comment]
> I think this is/could be caused by poor internet connection at the time.

I did indeed had intermittent network issues around the time I observed it

> if you see it happen look in the browser dev console for errors, as well as the network tab of the dev tools to see if the icomoon files failed to download.

will do, didn't think about at the time

**Comment by martonp** [comment]
> @martonp Note that there are cache busters on the icomoon font paths. I always forget about those, but I think you have an active PR that adds a glyph (or did we merge it already?), so should bump that.

I've bumped it in #1639.

**Comment by LasTshaMAN** [comment]
Just an update, encountered this recently once again (running on latest master, I'm also running dexc in Docker):

<img width="1344" alt="image" src="https://user-images.githubusercontent.com/7031569/173134531-bc77de23-8158-4abe-be9c-b65c43daafd5.png">

Under `console` tab there is:
```
The source list for the Content Security Policy directive 'connect-src' contains an invalid source: 'ws://[::]:5758'. It will be ignored.
```
Not sure what this one means, browser doesn't understand `[::]` ?
Googling a bit found this as well (could be related) https://github.com/w3c/webappsec-csp/issues/7

didn't manage to look under network tab though :( I accidentally refreshed the page,

maybe I just need a browser update ;)

**Comment by chappjc** [comment]
The network tab indicates it didn't even try to load the icomoon SVG file.  I don't know what's going on with your browsers there.

**Comment by chappjc** [comment]
Perhaps the csp error on the ws endpoint stops loading of the other network resources?

**Comment by chappjc** [comment]
> Not sure what this one means, browser doesn't understand `[::]` ?

Well, `[::]` is not a localhost address, that should be `[::1]`.  In live_test.go, the listener was fixed just two days ago: https://github.com/decred/dcrdex/blame/74114ac6f47298515378be37653b860ad7cd06f4/client/webserver/live_test.go#L1653
But I don't think this explains why the client (your browser) is using `[::]` to dial an endpoint with the `localhost` hostnamed.

Maybe try to use 127.0.0.X instead of `localhost` or `[::]` from now on to see if you ever run into the problem.

**Comment by chappjc** [comment]
Ah, that's why: https://github.com/decred/dcrdex/blob/74114ac6f47298515378be37653b860ad7cd06f4/client/webserver/webserver.go#L448

It is setting CSP to workaround a webkit/safari bug but it's injecting the webserver's address, which prior to the fix in https://github.com/decred/dcrdex/pull/1639 was `[::]` with live_test.go instead of an actual loopback address: https://github.com/decred/dcrdex/pull/1639

**Comment by LasTshaMAN** [comment]
Just an update to sum up my understanding of this issue so far,

I'm running `dexc` in Docker with webserver listening at `0.0.0.0:5758`:
```
docker run --rm -p 127.0.0.1:5758:5758 -v dcrdex_data:/root/.dexc user/dcrdex --webaddr=0.0.0.0:5758 --testnet
```
so `0.0.0.0` converts into `[::]` at this line  https://github.com/decred/dcrdex/blob/74114ac6f47298515378be37653b860ad7cd06f4/client/webserver/webserver.go#L444
and gets returned in `"Content-Security-Policy"` header where my Brave browser doesn't want to treat `[::]` as "any host allowed" (and throws error in console) -> which (likely) results into assets(pics) prevented (not always, only some times) from being loaded.

It looks like listening on `0.0.0.0` is the only way to expose ports from Docker-container to outside world on Mac (see [one](https://itecnote.com/tecnote/docker-app-server-ip-address-127-0-0-1-difference-of-0-0-0-0-ip/), [two](https://serverfault.com/questions/1084915/still-confused-why-docker-works-when-you-make-a-process-listen-to-0-0-0-0-but-no)).

I'll try using other browsers with this Docker setup to see whether its Brave-specific  quirk.

**Comment by ukane-philemon** [comment]
> although I think it's more about 0.0.0.0

I think so. I saw this issue again recently(several times) and only when I used `:8080` which translates to `[::]:8080` and result in the CSP error you mentioned earlier. It is not docker specific. We currently use `tcp` which will listen on both IPv4 and IPv6 addresses when an addr is not specified. (for IPv6 `0.0.0.0` translates to `[::]` but for IPv4 it remains the same).

We would want to keep support for both IP versions and find a workaround for  IPv6 translating `0.0.0.0` differently so the issue with icons turning into weird symbols can be resolved.
